### ISSUE TITLE
Can something be done with <$> (infix fmap from Control.Applicative)? Because it looks awful.

### ISSUE THREAD

**Issue by kindaro** [kosmikus/lhs2tex#58]
This must be an insignificant rant, but can something be done to improve this typesetting?

![swaygrab_1509108057-cut](https://user-images.githubusercontent.com/2599460/32104537-188e9336-bb4f-11e7-86cc-9428fcc37397.png)

It's way spread out and not nicely vertically aligned: the `$` character is way too high up relative to the angle brackets and the rest of the line.

Since `<$>` made it to Prelude, it sees more use as a nicer replacement for infix ``` `fmap` ```, so there must be quite a bunch of code around that would benefit from a better out-of-the-box typesetting.

Even just setting `<$>` in monospace font with some spacing from the surrounding code will make it look better (as you can see throughout this very message). To my knowledge, there's no nice and commonly accepted Unicode codepoint for this operator, but, for example, [here on github](https://github.com/tonsky/FiraCode) someone made a set of ligatures that features a nice `<$>` glyph.

**Comment by kosmikus** [comment]
You can just define a formatting directive and map it to anything you like.

**Comment by ivanperez-keera** [comment]
I agree. It does not look very good. In my opinion, the issues are both height and separation.

For a paper we wrote together, @turion created this rule:

https://github.com/ivanperez-keera/lhs2tex-haskell-operators/blob/master/document.lhs#L16

It looks like this:

![untitled](https://user-images.githubusercontent.com/1162320/37563191-c718287e-2a51-11e8-8ff1-33687f857d59.png)

I've created that repo to start collecting these formatting rules. We can keep it simple, just putting them in a versioned file that everyone can download regularly and import for all their projects.

**Comment by plredmond** [comment]
```latex
%format <$> = "\mathbin{\langle\$\rangle}"
%format <*> = "\mathbin{\langle*\rangle}"

**Comment by turion** [comment]
> https://github.com/ivanperez-keera/lhs2tex-haskell-operators/blob/master/document.lhs#L16

@ivanperez-keera do you know where this repository is now?

**Comment by ivanperez-keera** [comment]
No, but I have a copy of our paper and the rule you created was:

```latex
%format <$> = "\inanglebrackets{\raisebox{0.1pt}{\scalebox{.8}{\$}}}"
```
where
```latex
\newcommand{\inanglebrackets}[1]{\mathbin{<\hspace{-1.6pt}\mathclap{#1}\hspace{-1.6pt}>}}
```
Does that help?

**Comment by ivanperez-keera** [comment]
I'll leave that to you. For the most part, I stopped using lhs2TeX. When using extensions and such it always required too much tweaking and I had to modify my haskell to make it work.
